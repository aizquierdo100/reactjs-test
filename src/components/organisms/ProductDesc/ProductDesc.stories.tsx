import React, { useState } from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';
import ProductDesc from './ProductDesc';

export default {
    title: 'Components/Molecules/ProductDesc',
    component: ProductDesc,
    argTypes: {
        backgroundColor: { control: 'color' },
    },
} as ComponentMeta<typeof ProductDesc>;

const Template: ComponentStory<typeof ProductDesc> = (args) => <ProductDesc {...args} />

export const BaseProductDesc = Template.bind({});
BaseProductDesc.args = {
    image: 'model1.jpg',
    name: 'Stripe dress',
    description: 'Description'
};

