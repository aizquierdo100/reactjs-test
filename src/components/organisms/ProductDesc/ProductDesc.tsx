import React, {FC} from 'react';

import Image from "../../atoms/Image/Image";
import Label from "../../atoms/Label/Label";
import '../../../assets/styles/grid.scss';


interface IProps {
    image: string;
    name: string;
    description: string;
}

const ProductDesc: FC<IProps> = ({image, name, description}) => {
    return (
        <div className='container container--center container--wrap--m'>
            <div className='g--8'>
                <Image src={image} alt={name} zoom/>
            </div>
            <div className='g--8'>
                <Label>{name}</Label>
                <Label>{description}</Label>
            </div>
        </div>
    );
};

export default ProductDesc;