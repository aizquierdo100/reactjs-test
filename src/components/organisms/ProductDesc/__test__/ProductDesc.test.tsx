/**
 *
 * CategoryCard Test
 * @format
 *
 */

import { composeStories } from "@storybook/testing-react";
import * as Stories from "../ProductDesc.stories";
import { render } from "@testing-library/react";

const { BaseProductDesc } = composeStories(Stories);

describe("ProductDesc", () => {
    test("renders BaseProductDesc correctly", () => {
        const view = render(<BaseProductDesc />);
        expect(view).toMatchSnapshot();
    });
});
