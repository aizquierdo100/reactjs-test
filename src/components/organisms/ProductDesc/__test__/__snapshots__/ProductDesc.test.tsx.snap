// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductDesc renders BaseProductDesc correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container container--wrap--m"
      >
        <div
          class="g--8 g-m--12"
        >
          <img
            alt="Stripe dress"
            class="img-card"
            src="model1.jpg"
          />
        </div>
        <div
          class="g--4 g-m--12"
        >
          <p
            class="mb-2 title"
          >
            Stripe dress
          </p>
          <p>
            Description
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container container--wrap--m"
    >
      <div
        class="g--8 g-m--12"
      >
        <img
          alt="Stripe dress"
          class="img-card"
          src="model1.jpg"
        />
      </div>
      <div
        class="g--4 g-m--12"
      >
        <p
          class="mb-2 title"
        >
          Stripe dress
        </p>
        <p>
          Description
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
